docs/config.md

```diff
@@ -53,14 +53,29 @@ The main difference here is that we are composing patterns together. Every patte
 
 Will match nothing, because foo() and foo(1) can never occur together. If we made the first pattern `foo(...)`, the rule will work, although the pattern is unecessary.
 
-There are several operators that can be used under `patterns`:
+## Pattern Composition Operators
+
+### basic operators
+
+There are several operators that can be used in a rule. At the top level, there are three:
 
 - `pattern`: The rule will only fire if this pattern is found.
-- `pattern-not`: Opposite of `pattern`
-- `pattern-either`: You can put multiple other patterns under this; any of those patterns will count as a match.
-- `pattterns`: Like `pattern-either`, you can put multiple patterns under this to create nested, implicitly-ANDed instructions.
-- `pattern-inside`: The rule will only fire if the following patterns are inside this specified pattern. Useful for specifying a function that this behavior must occur in, for instance.
-- `pattern-not-inside`: Opposite of `pattern-inside`
+- `pattern-either`: (logical OR) Nest multiple other patterns under this; any of those patterns will count as a match.
```

Comment:
Would it make sense to use `patterns-or` and `patterns-and` here? These keys may be a bit more obvious to first-time users with some programming background. We can always keep the current keys and only document the new ones to ensure backward compatibility.

Reply:
@mschwager I am happy to change the key names but probably better to do it in a separate commit - feel free to comment on #148 !